dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsWriter.java

```diff
@@ -94,7 +94,12 @@ public static void preparePublicationFile(TCKResultsInfo resultInfo) {
             }
             output.write(adocContent);
             for (TestSuiteResult result : xmlParser.getResults()) {
-                output.write(result.toString());
+                // Checking if the result is for the correct repeat (avoiding adding wrong repeat results to wrong TCK report repeat files)
+                if (result.toString().contains(repeat)) {
```

Comment:
do you want a null check on the repeat string?

Reply:
does the code which adds the repeat string ever include a separator char after the test name?

Reply:
answered in https://github.com/OpenLiberty/open-liberty/pull/27429#pullrequestreview-1841044600